.rubocop.yml

```diff
@@ -200,3 +200,745 @@ Metrics/ParameterLists:
+  - app/controllers/admin/order_cycles_controller.rb
+  - app/controllers/admin/subscriptions_controller.rb
+  - app/controllers/checkout_controller.rb
+  - app/models/enterprise.rb
+  - app/models/order_cycle.rb
+  - app/models/product_import/entry_processor.rb
+  - app/models/product_import/entry_validator.rb
+  - app/models/product_import/product_importer.rb
+  - app/models/spree/ability_decorator.rb
+  - app/serializers/api/enterprise_serializer.rb
+  - app/services/cart_service.rb
+  - lib/open_food_network/bulk_coop_report.rb
+  - lib/open_food_network/enterprise_fee_calculator.rb
+  - lib/open_food_network/order_cycle_form_applicator.rb
+  - lib/open_food_network/order_cycle_management_report.rb
+  - lib/open_food_network/order_cycle_permissions.rb
+  - lib/open_food_network/orders_and_fulfillments_report.rb
+  - lib/open_food_network/packing_report.rb
+  - lib/open_food_network/payments_report.rb
+  - lib/open_food_network/permissions.rb
+  - lib/open_food_network/products_cache.rb
+  - lib/open_food_network/xero_invoices_report.rb
+
+Metrics/ModuleLength:
+  Max: 100
+  Exclude:
+  - lib/open_food_network/column_preference_defaults.rb
+  - spec/controllers/admin/enterprises_controller_spec.rb
+  - spec/controllers/admin/order_cycles_controller_spec.rb
+  - spec/controllers/api/order_cycles_controller_spec.rb
+  - spec/controllers/spree/api/products_controller_spec.rb
+  - spec/lib/open_food_network/address_finder_spec.rb
+  - spec/lib/open_food_network/customers_report_spec.rb
+  - spec/lib/open_food_network/enterprise_fee_calculator_spec.rb
+  - spec/lib/open_food_network/option_value_namer_spec.rb
+  - spec/lib/open_food_network/order_cycle_form_applicator_spec.rb
+  - spec/lib/open_food_network/order_cycle_permissions_spec.rb
+  - spec/lib/open_food_network/order_grouper_spec.rb
+  - spec/lib/open_food_network/permissions_spec.rb
+  - spec/lib/open_food_network/products_and_inventory_report_spec.rb
+  - spec/lib/open_food_network/products_cache_spec.rb
+  - spec/lib/open_food_network/proxy_order_syncer_spec.rb
+  - spec/lib/open_food_network/scope_variant_to_hub_spec.rb
+  - spec/lib/open_food_network/subscription_payment_updater_spec.rb
+  - spec/lib/open_food_network/tag_rule_applicator_spec.rb
+  - spec/lib/open_food_network/users_and_enterprises_report_spec.rb
+  - spec/models/spree/ability_spec.rb
+  - spec/models/spree/adjustment_spec.rb
+  - spec/models/spree/line_item_spec.rb
+  - spec/models/spree/payment_spec.rb
+  - spec/models/spree/product_spec.rb
+  - spec/models/spree/variant_spec.rb
+  - spec/support/request/web_helper.rb
+
+Metrics/ParameterLists:
+  Max: 5
+  Exclude:
+  - app/helpers/angular_form_builder.rb
+  - app/models/product_import/entry_processor.rb
+  - app/models/product_import/entry_validator.rb
+  - lib/open_food_network/xero_invoices_report.rb
+  - spec/features/admin/reports_spec.rb
+
+Metrics/BlockNesting:
+  Max: 3
+  Exclude:
+  - app/controllers/checkout_controller.rb
```

Comment:
How did you generate all these exclusions? It'd be great to doc it at the top of this file and also in a wiki page, if we are to maintain it ourselves.

Reply:
That was a manual process:
```
rubocop > rubo.log
# inspect log file to see which cops are failing
# copy cop configurations and add Exclude parameter
grep ExampleCop rubo.log | cut -d ":" -f 1 | sort -u >> .rubocop.yml
# use vim to add `- ` before each line
```

Do you think this process is valuable enough to include it in a comment?

Reply:
I'd say so. The easier we make it for other devs, the greater the chances to have it up-to-date.